### DISCUSSION TITLE
Koreader is not saving the page on exit.

### DISCUSSION THREAD

**Discussion by Zeyadas** [koreader/koreader#8328] [Q&A] [answered]
Koreader version: 9.2021
Device: Samsung galaxy Tab A7" (T-285)
I believe it's something in the settings I did and didn't figure out what I exactly did, and I didn't post this in issue section because it's not an issue..
So, when reading a book I'm on page 286 for example, when I close the app and reopen it, it starts from the beginning of the book.
Any ideas on how to solve this problem?

**Comment by Zeyadas** [comment]
[Update], that happens in a series of books (an Encyclopedia, scanned pdf documents), all it's files didn't save reading progress on exit. Maybe it's something related to the documents itself.
And by the way on this device I noticed that koreader is not creating the annoying (.sdr) folders for all my pdfs here.

**Comment by Zeyadas** [comment]
[Update 2]:
> that happens in a series of books (an Encyclopedia, scanned pdf documents),

On my second tablet a Samsung galaxy Tab A 8" with 2021.9-37 koreader version 
Same files, page progress works perfectly (saving and back normally to the page I left) and the (.sdr) folder is showing up

**Comment by mergen3107** [comment]
Recently there was a thread where OP had such problems because of SD card. AFAIK KOReader can’t write on SD cards

**Reply by Zeyadas** [reply]
Moved this exact file to device memory, worked perfectly and saved the page progress, and (.sdr) file is created on device memory. So it's writing on SD card issue as you said. 
Thank you.

**Comment by poire-z** [comment, marked as answer]
Probably a writing-permission issue on this SDCard (with all the Android access/permissions crap, can't say more).
But when there is this, koreader save the settings in koreader/history/ (which is some old behaviour, that happens to still work when it can't create the .sdr/ directory, but with many issues - we discussed this elsewhere https://github.com/koreader/koreader/issues/8281#issuecomment-944340762).

BUT, you must be in this strange situation that:
- you had previously some .sdr/metadata.xxx.lua created alongside the book on the SDcard - so you didn't have any permission/writting problem at the time
- KOReader can't update/write it NOW, so it _may_ (not sure) write the book settings in koreader/history/
- on re-opening, .sdr/metadata.xxx.lua still exists and is readable - even if it's an old one; KOReader will prefer it and use it - and it will ignore (and not even check) the recent one in koreader/history/

Solutions:
- fix this permission problem (if that's even possible)
- or remove the .sdr/ folders with another app (if you have files in koreader/history/, they then should be used - if not, well, you lost your settings and positions :/)

**Reply by Zeyadas** [reply]
I'll try that, thank you.
